Feature-1678: Doc Updates Part 2

doulikecookiedough opened the pull request:
Greetings @artntek, when you have scope - can you please help review my changes to the admin docs for `install.rst`, `contributors.rst`, `dataone.rst` and `themes.rst`?
- The main document that has been heavily modified is `install.rst`

**Summary of Changes:**
- Added and revised some css styles in `metacatui.css` to improve how the docs are displayed
- Removed the `index` option in the upper righthand corner (so it's only `next`, and `previous` now) (modified `layout.html`)
    - Commented out `searchbox` related code that was causing misleading console errors
- RE: `install.rst`
    - Added `rabbitmq` section, and updated `solr` section (including the notes regarding solr 9.5 & the allowPaths having to be a real path, and how Java 1.8 must be used during upgrade to 2.19.0)
    - Added recommendations for solr 9.5.0, tomcat9 and postgres14.
    - Added `Upgrading to Metacat 3.0.0` underneath `Upgrade Metacat` section
    - Formatted the document's usage of spacing so that it's consistent all throughout
    - Updated screenshots, fixed typos and slightly revised wording throughout
- RE: `contributors.rst`
    - Updated this section to match the new 3.0.0 README.md
- RE: `themes.rst`
   - Updated screenshot
   - Removed skins references
- RE: `dataone.rst`
   - Removed deprecated section on replication (obsolete)
   - Fixed typos, updated screenshot
- RE: `intro.rst`
   - Combined the sentences related to `metacatui` into one point in 'Metacat features'

artntek reviewed (changes requested):
Is it possible to build and deploy the docs somewhere that I can see them rendered? Do you have a VM, for example?

doulikecookiedough commented on docs/user/metacat/source/contributors.rst:
```diff
@@ -6,17 +6,23 @@ open source project.  Main developers and additional patch contributors are
-Contributors
+Current Contributors
+------------
+  - Jing Tao (tao@nceas.ucsb.edu) [ORCID: 0000-0002-1209-5268](http://orcid.org/0000-0002-1209-5268)
+  - Matthew Brooke (brooke@nceas.ucsb.edu) [ORCID: 0000-0002-1472-913X](http://orcid.org/0000-0002-1472-913X)
+  - Dou Mok (mok@nceas.ucsb.edu) [ORCID: 0000-0002-6076-8092](http://orcid.org/0000-0002-6076-8092)
+  - Matthew B. Jones (jones@nceas.ucsb.edu): [ORCID: 0000-0003-0077-4738](http://orcid.org/0000-0003-0077-4738)
```
I've amended this, now this matches with the `README.md`

doulikecookiedough commented on docs/user/metacat/source/dataone.rst:
```diff
@@ -307,20 +318,6 @@ or can be set using the `CNReplication.setReplicationPolicy`_ service.
 .. _CNReplication.setReplicationPolicy: http://releases.dataone.org/online/d1-architecture-1.0.0/apis/CN_APIs.html#CNReplication.setReplicationPolicy
 
 
-Generating DataONE System Metadata
```
I synced up with @taojing2002 that this section is for the old Metacat API for replications, which has been removed. I will double check with him that I didn't delete too much, or if we should retain any of the content.

doulikecookiedough replied:
I connected with @taojing2002 and we have determined to restore this section but with a note:

> **Note:** The following feature is not available in Metacat 3.0.0. If you are upgrading your Metacat from Metacat version 1.*, we recommend that you first upgrade to 2.19.0 before proceeding with generating DataONE System Metadata.

doulikecookiedough commented on docs/user/metacat/source/themes.rst:
```diff
@@ -1,21 +1,18 @@
-|
-
 Metacat's theming system, `MetacatUI <https://github.com/NCEAS/metacatui>`_, is deployed
 separately from Metacat, allowing more independent user interface customization. Check the
```
Sorry @artntek, do you mean to move the screenshot above the second paragraph?

artntek replied:
no - I meant replace:

> "allowing more independent user interface customization"

with:

> "allowing more independent visual customization"

...to avoid having "user interface" twice in the same sentence

doulikecookiedough commented on docs/user/metacat/source/themes/metacatui/layout.html:
```diff
@@ -124,15 +127,15 @@ <h3>{{ _('Quick search') }}</h3>
-        HAS_SOURCE:  {{ has_source|lower }}
-      };
-    </script>
+<!--    <script type="text/javascript">-->
+<!--      var DOCUMENTATION_OPTIONS = {-->
+<!--        URL_ROOT:    '{{ pathto("", 1) }}',-->
+<!--        VERSION:     '{{ release|e }}',-->
+<!--        COLLAPSE_MODINDEX: false,-->
+<!--        FILE_SUFFIX: '{{ file_suffix }}',-->
+<!--        HAS_SOURCE:  {{ has_source|lower }}-->
+<!--      };-->
+<!--    </script>-->
```
I've removed the commented out code - and note, this code that was commented out was to resolve console error:

```
Uncaught SyntaxError: Identifier 'DOCUMENTATION_OPTIONS' has already been declared (at documentation_options.js:1:1)
```

doulikecookiedough commented on docs/user/metacat/source/install.rst:
```diff
@@ -33,16 +37,16 @@ running correctly:
-* 200 MB disk space (Note: The amount of disk space required depends on the size of your RDBMS tablespace and the the size and number of documents stored. Metacat itself requires only about 140 MB of free space after installation).
+  * at least 512MB RAM
+
+  * 200 MB disk space (Note: The amount of disk space required depends on the size of your RDBMS tablespace and the the size and number of documents stored. Metacat itself requires only about 140 MB of free space after installation).
```
When checking the installed location (`var/metacat`), it's approximately 189MB (without too many datasets or metadata). I'll omit the second sentence so we don't have to keep updating it.

doulikecookiedough commented on docs/user/metacat/source/install.rst:
```diff
@@ -9,19 +7,25 @@ Instructions for the Linux system is included in this section.
 In addition to meeting the recommended system requirements, the server on which
-you wish to install Metacat must have the following software installed and
-running correctly:
+you wish to install Metacat must have the following software installed and running correctly:
```
Hmm, I would like to reword it as such instead of specifying a version or adding the sentence to see each section.

> In addition to meeting the recommended system requirements, the server on which you wish to install **the latest version of Metacat** must have the following software installed and running correctly

I have adjusted this and please let me know if you think this is not clear enough (and if so, I think adding the specific version number is better).

doulikecookiedough commented on docs/user/metacat/source/install.rst:
```diff
@@ -59,16 +63,25 @@ For the impatient or those who have already installed Metacat and know what
-8. ``sudo cp <metacat_package_dir>/metacatui.war <tomcat_app_dir>``
-9. ``sudo /etc/init.d/tomcat7 restart``
-10. Configure Metacat through the Web interface
+  1. Download and install prerequisites (`Java 17`_, `Apache Tomcat`_ 7, PostgreSQL_, `RabbitMQ`_, `Solr Server`_, `Apache HTTPD Server`_)
```
Nice catch, thank you!

doulikecookiedough commented on docs/user/metacat/source/install.rst:
```diff
@@ -704,208 +795,138 @@ with Metacat's Authorization Configuration screen. Note that if you do not have
+   downloaded Metacat. Note: Throughout the instructions, we will refer to this
+   directory as ``<metacat_src_dir>``. 
+
+  * Set the build.tomcat.dir property to your Tomcat installation directory.
```
I have removed that sentence.

doulikecookiedough commented on docs/user/metacat/source/install.rst:
```diff
@@ -467,61 +519,101 @@ The Metacat servlet automatically creates the required database schema. For
 
-Metacat only supports ``Solr 8.*``, ``NOT 9.*``. You may download the binary releases from:
+Starting from v2.13.0, Metacat uses the external Solr HTTP server as the search engine. Unfortunately
+the Solr Debian packages that come with the Ubuntu operating system are obsoleted, so you will have
```
Yes - I believe it's still true. I do not see any `solr` debian packages when searching through the debian website:
- https://packages.debian.org/search?searchon=sourcenames&keywords=solr

When switching to `package-names` to search, I do see some old solr packages but they are very old (`3.6.2`).
- https://packages.debian.org/search?suite=default&section=all&arch=any&searchon=names&keywords=solr

doulikecookiedough commented on docs/user/metacat/source/dataone.rst:
```diff
@@ -82,14 +88,19 @@ allow Member Nodes to easily interact with one another and to provide a unified
+
+  * Resolution service to map unique identifiers to the Member Nodes that hold data
+
+  * Authentication against a shared set of accounts based on CILogon_ and InCommon_
```
I have revised the second point to refer to ORCID instead and confirmed that the first point:
> Matt: the first one is still true -- this is the `resolve` api service

doulikecookiedough commented:
Greetings @Matthew Brooke - Thank you for reviewing part 2 of my admin doc updates! I believe I have addressed the issues you commented, and left the ones that may still need to be updated unresolved. Please let me know if you have any other feedback.

doulikecookiedough commented on docs/user/metacat/source/install.rst:
```diff
@@ -46,7 +46,7 @@ System requirements for running Metacat:
   * at least 512MB RAM
 
-  * 200 MB disk space (Note: The amount of disk space required depends on the size of your RDBMS tablespace and the the size and number of documents stored. Metacat itself requires only about 140 MB of free space after installation).
+  * 200 MB disk space (Note: The amount of disk space required depends on the size of your RDBMS tablespace and the the size and number of documents stored.).
```
I have omitted this

doulikecookiedough commented:
Thank you for reviewing this branch again @artntek - I have pushed a few commits to update the incorrect link to ORCID in `dataone.rst`, updated the wording in `themes.rst` and removed the system requirement point in `install.rst` that feels like will need to be updated frequently.
